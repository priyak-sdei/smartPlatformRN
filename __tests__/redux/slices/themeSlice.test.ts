import themeReducer, { setTheme } from '../../../src/redux/slices/themeSlice';

describe('themeSlice', () => {
    it('should return the initial state', () => {
        expect(themeReducer(undefined, { type: 'theme/unknown' })).toEqual({
            mode: 'light',
        });
    });

    it('should handle setTheme', () => {
        const previousState = { mode: 'light' as 'light' };
        expect(themeReducer(previousState, setTheme('dark'))).toEqual({ mode: 'dark' });
    });

    it('matches snapshot', () => {
        expect(themeReducer(undefined, { type: 'theme/unknown' })).toMatchSnapshot('themeSlice initial state');
    });
}); 