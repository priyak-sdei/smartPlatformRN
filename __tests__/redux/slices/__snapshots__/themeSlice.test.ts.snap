// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themeSlice matches snapshot: themeSlice initial state 1`] = `
{
  "mode": "light",
}
`;
